client/jetpack-cloud/sections/agency-dashboard/sidebar/index.tsx

```diff
@@ -1,102 +1,196 @@
+		page.redirect( path );
 	};
 
-	const isPluginsPage = path.includes( '/plugins' );
-	const isPluginManagementEnabled = config.isEnabled( 'jetpack/plugin-management' );
+	const isPartnerPortal = path.startsWith( '/partner-portal' );
 
-	const isAtomicSiteCreationEnabled = config.isEnabled(
-		'jetpack/pro-dashboard-wpcom-atomic-hosting'
-	);
+	const SidebarItem = ( { className, icon, withChevron = false, suffix, children, ...props } ) => {
+		return (
+			<Item
+				className={ classnames( className, {
+					'is-active': path === props.link,
+					'sidebar-button': true,
+				} ) }
+				{ ...props }
+			>
+				<HStack justify="flex-start">
+					{ icon && <Icon style={ { fill: 'currentcolor' } } icon={ icon } size={ 24 } /> }
+					<FlexBlock>{ children }</FlexBlock>
+					{ withChevron && <Icon icon={ chevronRightSmall } size={ 24 } /> }
+					{ ! withChevron && suffix }
+				</HStack>
+			</Item>
+		);
+	};
+
+	const initialPath = isPartnerPortal ? '/partner-portal' : '/';
 
 	return (
-		<div>
-			<SiteSelector
-				showAddNewSite
-				showAllSites
-				isJetpackAgencyDashboard={ isAtomicSiteCreationEnabled }
-				className="sidebar__site-selector"
-				allSitesPath={ path }
-				siteBasePath="/backup"
-				wpcomSiteBasePath={ isAtomicSiteCreationEnabled && 'https://wordpress.com/home' }
-			/>
-			<Sidebar className="sidebar__jetpack-cloud">
-				<SidebarRegion>
-					<CurrentSite
-						// Ignore type checking because TypeScript is incorrectly inferring the prop type due to JSX usage in <CurrentSite /> component
-						// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-						// @ts-ignore
-						forceAllSitesView
-					/>
-					<SidebarMenu>
-						<SidebarItem
-							customIcon={ <JetpackIcons icon="dashboard" /> }
-							label={ translate( 'Dashboard', {
-								comment: 'Jetpack sidebar navigation item',
-							} ) }
-							link="/dashboard"
-							onNavigate={ onNavigate( 'Jetpack Cloud / Dashboard' ) }
-							selected={ ! isPluginsPage }
-						/>
-						{ isPluginManagementEnabled && (
-							<SidebarItem
-								customIcon={ <Icon className="sidebar__menu-icon" size={ 28 } icon={ plugins } /> }
-								label={ translate( 'Plugins', {
-									comment: 'Jetpack sidebar navigation item',
-								} ) }
-								link="/plugins/manage"
-								onNavigate={ onNavigate( 'Jetpack Cloud / Plugins' ) }
-								selected={ isPluginsPage }
-							/>
-						) }
-					</SidebarMenu>
-				</SidebarRegion>
-				<SidebarFooter>
-					<SidebarMenu>
-						<SidebarItem
-							label={ translate( 'Get help', {
-								comment: 'Jetpack Cloud sidebar navigation item',
-							} ) }
-							link="https://jetpack.com/support"
-							className="sidebar__jetpack-cloud-item-has-border"
-							customIcon={ <JetpackIcons icon="help" /> }
-							onNavigate={ onNavigate( 'Jetpack Cloud / Support' ) }
-						/>
-					</SidebarMenu>
-				</SidebarFooter>
-			</Sidebar>
-		</div>
+		<>
+			<NavigatorProvider
```

Comment:
I'm having some doubts if we are using the `NavigatorProvider` correctly here.  Shouldn't we wrap the entire layout with the `NavigatorProvider` so we properly inherit the navigation states and transitions? From what I can tell, the transition effect only happens within the sidebar, not the entire page.


<img width="1724" alt="Screen Shot 2023-09-26 at 4 02 40 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/db536d80-c247-4241-988b-d3b32a434914">

Reply:
Although looking at the current routing implementation, I'm having difficulty thinking how this `NavigationProvider` fits in the Page context. 

It looks like we will have to [go deep](https://github.com/Automattic/wp-calypso/blob/trunk/client/controller/index.node.js#L51) if we are to integrate it.

Reply:
> I'm having some doubts if we are using the NavigatorProvider correctly here. Shouldn't we wrap the entire layout with the NavigatorProvider so we properly inherit the navigation states and transitions? From what I can tell, the transition effect only happens within the sidebar, not the entire page.

I'll take a look at this.

> Although looking at the current routing implementation, I'm having difficulty thinking how this NavigatorProvider fits in the Page context.

You can see how it is implemented here and see if you find something: https://github.com/WordPress/gutenberg/blob/9fcfaac513297b0517cf33c696a59d2324dd8e1e/packages/edit-site/src/components/sidebar/index.js#L75

Reply:
`NavigatorProvider` will only affect the sidebar in this case, this the overall routing is handled differently (via `page` it seems)